src/Caller.h

```diff
@@ -123,7 +128,79 @@ class Caller {
   virtual void processPeptideScores(Scores& allScores) {}
   virtual void processProteinScores(ProteinProbEstimator* protEstimator) {}
 #endif
+
+bool detect_tab(std::string file_name)
+{
+    // open C++ stream to file
+    std::ifstream file(file_name.c_str());
+    // file not opened, return false
+    if(!file.is_open()) return false;
+    // read a line from the file       
+    std::string wtf;
+    std::istream &in= std::getline(file, wtf);
+    // unable to read the line, return false
+    if(!in) return false;
+    // try to find a '\t', return true if '\t' is found within the string
+    return std::find(wtf.begin(), wtf.end(), '\t')!= wtf.end();
+}
+
+std::string detect_decoy_prefix(std::string file_name)
```

Comment:
Place this as a (static?) method of a Class. Possibly TabReader.

Reply:
I'm not sure if we want it to be static since after I moved it to DataSet: https://github.com/percolator/percolator/commit/05b0a8aa6893b4f4197c3ec74e22ea11baf19c98